packages/samples/headless-react/src/components/breadcrumb-manager/breadcrumb-manager.class.tsx

```diff
@@ -0,0 +1,145 @@
+    this.unsubscribe();
+  }
+
+  private updateState() {
+    this.setState(this.controller.state);
+  }
+
+  private renderFacetBreadcrumbs() {
+    return (
+      <li>
+        {this.state.facetBreadcrumbs.map((facet) => (
+          <span key={facet.facetId}>
+            {facet.field}:{' '}
+            {facet.values.map((breadcrumb) => (
+              <button
+                key={breadcrumb.value.value}
+                onClick={() => breadcrumb.deselect()}
+              >
+                {breadcrumb.value.value}
+              </button>
+            ))}
+          </span>
+        ))}
+      </li>
+    );
+  }
+
+  private renderCategoryFacetBreadcrumbs() {
+    return (
+      <li>
+        {this.state.categoryFacetBreadcrumbs.map((facet) => (
+          <span key={facet.path.join('/')}>
+            {facet.field}:{' '}
+            <button onClick={() => facet.deselect()}>
+              {facet.path.map(({value}) => value).join('/')}
+            </button>
+          </span>
+        ))}
+      </li>
+    );
+  }
+
+  private getKeyForRange(value: NumericFacetValue | DateFacetValue) {
+    return `[${value.start}..${value.end}${value.endInclusive ? ']' : '['}`;
+  }
+
+  private formatNumericRangeValue(value: number) {
+    return this.props.numericFormat(value);
+  }
+
+  private renderNumericFacetBreadcrumbs() {
+    return (
+      <li>
+        {this.state.numericFacetBreadcrumbs.map((facet) => (
+          <span key={facet.facetId}>
+            {facet.field}:{' '}
+            {facet.values.map((breadcrumb) => (
+              <button
+                key={this.getKeyForRange(breadcrumb.value)}
+                onClick={() => breadcrumb.deselect()}
+              >
+                {this.formatNumericRangeValue(breadcrumb.value.start)} to{' '}
+                {this.formatNumericRangeValue(breadcrumb.value.end)}
+              </button>
+            ))}
+          </span>
+        ))}
+      </li>
+    );
+  }
+
+  private formatDate(dateStr: string) {
+    return parseDate(dateStr).format('MMMM D YYYY');
+  }
+
+  private renderDateFacetBreadcrumbs() {
+    return (
+      <li>
+        {this.state.dateFacetBreadcrumbs.map((facet) => (
+          <span key={facet.facetId}>
+            {facet.field}:{' '}
+            {facet.values.map((breadcrumb) => (
+              <button
+                key={this.getKeyForRange(breadcrumb.value)}
+                onClick={() => breadcrumb.deselect()}
+              >
+                {this.formatDate(breadcrumb.value.start)} to{' '}
+                {this.formatDate(breadcrumb.value.end)}
+              </button>
+            ))}
+          </span>
+        ))}
+      </li>
+    );
+  }
+
+  render() {
+    if (!this.state?.hasBreadcrumbs) {
+      return null;
+    }
+
+    return (
+      <ul>
+        {this.renderFacetBreadcrumbs()}
+        {this.renderCategoryFacetBreadcrumbs()}
+        {this.renderNumericFacetBreadcrumbs()}
+        {this.renderDateFacetBreadcrumbs()}
```

Comment:
I want to get opinions on showing one/two types of breadcrumbs instead of four?

If the goal is to have something that can be copy-pasted and work with all the facets, then having all four is useful.

If the goal is to teach, then I think having just one, maybe two is better. My concern is that seeing such a long sample will be intimidating to people exploring headless.

Reply:
I had a bit of confusion regarding the breadcrumbs state while writing this sample, especially since it's very different for facet and category, so I would vote for keeping these two.

Concerning numeric and date facet, I think they are pretty self-explanatory when looking at the other two and we could remove them.

Reply:
That's a fair point 🤔 

Another approach would be to have one "breadcrumb manager" sample for each type of facet ? 

But that could also invite to an anti pattern, where users would think that's the "way to go".

I think my vote would be to just have a sample for regular facet.